Sources/StreamChatUI/Utils/UIImageView+Extensions.swift

```diff
@@ -79,3 +79,36 @@ private extension UIImageView {
         set { objc_setAssociatedObject(self, &Self.nukeLoadingTaskKey, newValue, .OBJC_ASSOCIATION_RETAIN) }
     }
 }
+
+extension ImageProcessors {
+    public struct LateResize: ImageProcessing {
+        private var size: CGSize {
+            var size: CGSize = .zero
+            DispatchQueue.main.sync { size = sizeProvider() }
+            return size
+        }
+
+        private let sizeProvider: () -> CGSize
+
+        public init(sizeProvider: @escaping () -> CGSize) {
+            self.sizeProvider = sizeProvider
+        }
+
+        public func process(_ image: PlatformImage) -> PlatformImage? {
+            let size = self.size
+            guard size != .zero else { return image }
+            
+            return ImageProcessors.Resize(
+                size: size,
+                unit: .points,
+                contentMode: .aspectFill,
+                crop: true,
+                upscale: false
+            ).process(image)
+        }
+
+        public var identifier: String {
+            "com.github.kean/nuke/lateResize"
```

Comment:
Shouldn't we make this unique every time we create a processor?

Reply:
I was trying to figure out if it matters or not. Nuke docs don't say anything and the original `Resize` is adding just the parameters of the processor. If you create two `Resize` processors with the same size parameter, they will have the same identifier. Since we have size closure and getting the size is waiting synchronously for the main queue (may be expensive), I decided to remove it.